hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -1256,26 +1243,18 @@ protected void updateAggregateStatistics(List<HostState> hosts) {
     }
 
     private HostScore calculateHostScore(HostState host, double clusterMaxMeanActive, double clusterMaxDiskUsed) {
-        double score = 0;
         double meanActive = host.getMeanActiveTasks();
-        double usedDiskPercent = getUsedDiskPercent(host);
-        // If either metric is zero across the whole cluster, treat every host as having full load in that aspect
-        if (clusterMaxMeanActive <= 0) {
-            meanActive = 1;
-            clusterMaxMeanActive = 1;
-        }
-        if (clusterMaxDiskUsed <= 0) {
-            usedDiskPercent = 1;
-            clusterMaxDiskUsed = 1;
-        }
+        double diskUsedPercentModified = host.getDiskUsedPercentModified(config.getMinFreeDiskSpaceToRunJobs());
         int activeTaskWeight = config.getActiveTaskWeight();
         int diskUsedWeight = config.getDiskUsedWeight();
         // Assemble the score as a combination of the mean active tasks and the disk used
-        score += activeTaskWeight * Math.pow(meanActive / clusterMaxMeanActive, 2.5);
-        score += diskUsedWeight * Math.pow(usedDiskPercent / clusterMaxDiskUsed, 2.5);
+        double exponent = 2.5;
+        double score = activeTaskWeight * Math.pow(meanActive, exponent);
+        double diskPercentPowered = Math.pow(diskUsedPercentModified, exponent);
+        score += diskUsedWeight * diskPercentPowered;
         // If host is very full, make sure to give the host a big score
-        score = Math.max(score, (activeTaskWeight + diskUsedWeight) * usedDiskPercent);
-        return new HostScore(meanActive, usedDiskPercent, score);
+        score = Math.max(score, (activeTaskWeight + diskUsedWeight) * diskPercentPowered);
```

Comment:
Old code uses `usedDiskPercent` while new code uses `diskPercentPowered`. Is that intentional?